src/NuGetForUnity/Editor/Ui/NugetPreferences.cs

```diff
@@ -92,6 +92,17 @@ public override void OnGUI([CanBeNull] string searchContext)
                 ConfigurationManager.NugetConfigFile.Verbose = verbose;
             }
 
+            var slimRestore = EditorGUILayout.Toggle(
+                new GUIContent(
+                    "Slim Restore",
+                    "When slim restore is enabled, installing dependencies and checking against Unity pre-imported libs will be skipped on automatic restore. Manually restoring via menu option will ignore this setting."),
```

Comment:
```suggestion
                    "Slim restore is a faster way of installing / restoring packages after opening the Unity project. To achieve this the package installation step skips installing dependencies not listed inside the package.config, also it skips checking against Unity pre-imported libraries. If you have a complex project setup with multiple target platforms that include different packages it could be that you need to disable this feature. Manually restoring via menu option will ignore this setting."),
```
Not perfect but I think it is more understandable for users not "into it", provide some information so he can decide if he should enable / disable the feature.

Reply:
Agreed, I modified your suggestion a bit and split it into more rows for IDE readability.